test/test_bar.py

```diff
@@ -95,3 +95,25 @@ def test_bar_datazoom_inside():
     assert ': "inside"' in bar._repr_html_()
     assert ': "slider"' not in bar._repr_html_()
     bar.render()
+
+
+def test_bar_datazoom_both():
```

Comment:
it is better that once test case test one thing. there are two test sections here. better to split them up.